integration-test-groups/aws2/aws2-s3/src/test/java/org/apache/camel/quarkus/component/aws2/s3/it/Aws2S3Test.java

```diff
@@ -164,47 +164,51 @@ public void testKms() throws Exception {
         final String oid = UUID.randomUUID().toString();
         final String blobContent = "Hello KMS " + oid;
 
-        // Create
-        RestAssured.given()
-                .contentType(ContentType.TEXT)
-                .body(blobContent)
-                .post("/aws2/s3/object/" + oid + "?useKms=true")
-                .then()
-                .statusCode(201);
-
-        // Read
-        RestAssured.get("/aws2/s3/object/" + oid + "?useKms=true")
-                .then()
-                .statusCode(200)
-                .body(is(blobContent));
+        try {
+            // Create
+            RestAssured.given()
+                    .contentType(ContentType.TEXT)
+                    .body(blobContent)
+                    .post("/aws2/s3/object/" + oid + "?useKms=true")
+                    .then()
+                    .statusCode(201);
 
-        // Delete
-        deleteObject(oid);
+            // Read
+            RestAssured.get("/aws2/s3/object/" + oid + "?useKms=true")
+                    .then()
+                    .statusCode(200)
+                    .body(is(blobContent));
+        } finally {
+            // Delete
+            deleteObject(oid);
```

Comment:
Should be done in an AfterEach method instead of a  finally block

Reply:
The oid is a random generated id and there are some other tests such as ```copyObjectDeleteBucket``` which has a different test logic. There's also a ```@AfterEach``` method to check the objects leaking.

Reply:
@essobedo thanks for the review! 
We were hit by high AWS bills in the past when tests were either missing cleanup code or when something failed midair and the cleanup code was not invoked. Using try-catch helps to address both: missing cleanup code is easier to spot and existing cleanup code is harder to skip. Thus +1 for this change from my side.